ado_asana_sync/sync/sync.py

```diff
@@ -361,22 +361,22 @@ def sync_project(a: App, project):
         # Get the work item from the ID
         ado_task = a.ado_wit_client.get_work_item(wi.target.id)
 
-        # Skip this item if is not assigned, or the assignee does not match an Asana user.
+        # Skip this item if is not assigned, or the assignee does not match an Asana user,
+        # unless it has previously been matched.
+        existing_match = TaskItem.find_by_ado_id(a, ado_task.id)
         ado_assigned = get_task_user(ado_task)
-        if ado_assigned is None:
+        if ado_assigned is None and existing_match is None:
             logging.debug(
                 f"{ado_task.fields['System.Title']}:skipping item as it is not assigned"
             )
             continue
         asana_matched_user = matching_user(asana_users, ado_assigned)
-        if asana_matched_user is None:
+        if asana_matched_user is None and existing_match is None:
             continue
 
-        # Check if this is an already mapped item.
-        item = TaskItem.find_by_ado_id(a, ado_task.id)
-        if item is None:
+        if existing_match is None:
             logging.info(f"{ado_task.fields['System.Title']}:unmapped task")
-            item = TaskItem(
+            existing_match = TaskItem(
                 ado_id=ado_task.id,
                 ado_rev=ado_task.rev,
                 title=ado_task.fields["System.Title"],
```

Comment:
**Suggestion:** Consider refactoring the code to reduce the complexity of the 'sync_project' function. The function is too long and contains too many conditional statements, which makes it hard to understand and maintain. You can extract some of the logic into separate functions.
```suggestion
        def sync_project(a: App, project):
            # Get the work item from the ID
            ado_task = a.ado_wit_client.get_work_item(wi.target.id)
            existing_match = find_or_create_task_item(a, ado_task)
            if should_skip_item(ado_task, existing_match):
                continue
            asana_matched_user = matching_user(asana_users, get_task_user(ado_task))
            if should_skip_item_based_on_user(asana_matched_user, existing_match):
                continue
            update_task_item(existing_match, ado_task, asana_matched_user)
```